### PR TITLE
Add folder for movement licence and addmocha retry

### PR DESCRIPTION
**Pull request by eoin-corr-git**

### CODE REVIEW

File: wdio.conf.js [hunk 1]
```diff
@@ -136,7 +136,8 @@ export const config = {
   // See the full list at http://mochajs.org/
   mochaOpts: {
     ui: 'bdd',
-    timeout: debug ? 24 * 60 * 60 * 1000 : 60000
+    timeout: debug ? 24 * 60 * 60 * 1000 : 60000,
+    retries: 1
   }
```
Comment (hughfdjackson) [inline wdio.conf.js:141]: This part of the change looks good!

File: test/specs/noJavascript/landing.spec.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { browser } from '@wdio/globals'
 
-import landingPage from '~/test/page-objects/landingPage'
-import toFromFarmPage from '~/test/page-objects/toFromFarmPage'
+import landingPage from '~/test/page-objects/movementLicence/landingPage'
+import toFromFarmPage from '~/test/page-objects/movementLicence/toFromFarmPage'
```
Comment (hughfdjackson) [inline test/specs/noJavascript/landing.spec.js:4]: I think I preferred it before - unless there's a practical reason for the restructure?

File: .github/workflows/run-tests-browserstack.yaml [hunk 3]
```diff
@@ -2,7 +2,12 @@ name: Browserstack Tests
 
 on:
   workflow_dispatch: # Manually trigger the suite
-  workflow_call:     # Trigger it from another project
+  workflow_call:   
+    secrets:
+      BROWSERSTACK_USER:
+        required: true
+      BROWSERSTACK_KEY:
+        required: true  # Trigger it from another project
```
Comment (hughfdjackson) [inline .github/workflows/run-tests-browserstack.yaml:10]: I didn't know you could do this !

**Review by hughfdjackson — CHANGES_REQUESTED** [review]

File: test/specs/noJavascript/landing.spec.js (hunk 2 above)
Comment (eoin-corr-git) [reply test/specs/noJavascript/landing.spec.js:4]: Should help when navigating the tests when we add more journeys, just a natural grouping

File: .github/workflows/run-tests-browserstack.yaml (hunk 3 above)
Comment (eoin-corr-git) [reply .github/workflows/run-tests-browserstack.yaml:10]: Neither did I until yesterday 😁

File: test/specs/parishHolding.spec.js [hunk 4]
```diff
@@ -72,8 +76,12 @@ describe('Paris holding page test', () => {
     await ParishHoldingNumberPage.inputParishHoldingNumberAndContinue(
       validInput
     )
-    await expect(ParishHoldingNumberPage.cphInputFieldError).not.toBeDisplayed()
+    await expect(
+      await ParishHoldingNumberPage.cphInputFieldError()
+    ).not.toBeDisplayed()
     await ParishHoldingNumberPage.selectBackLink()
-    await expect(ParishHoldingNumberPage.cphNumberInput).toHaveValue(validInput)
+    await expect(await ParishHoldingNumberPage.cphNumberInput()).toHaveValue(
```
Comment (hughfdjackson) [inline test/specs/parishHolding.spec.js]: Do we need to await the expect, given that there's already any await within the arguments?

File: test/page-objects/parishHoldingNumberPage.js [hunk 5]
```diff
@@ -12,37 +12,40 @@ class ParishHoldingNumberPage extends Page {
     return 'What is the County Parish Holding (CPH) number of your farm or premises where the animals are moving off?'
   }
 
-  get cphNumberInput() {
-    return $('#cph-number')
+  async cphNumberInput() {
```
Comment (hughfdjackson) [inline test/page-objects/parishHoldingNumberPage.js:15]: No need to mark these as async if we're returning a synchronous result.

File: test/page-objects/parishHoldingNumberPage.js [hunk 6]
```diff
@@ -12,37 +12,40 @@ class ParishHoldingNumberPage extends Page {
...
+    return 'Enter the CPH number in the correct format, for example, 12/345/6789'
   }
 
   get noInputError() {
     return 'Enter the farm or premises CPH number'
   }
 
   async inputParishHoldingNumberAndContinue(text) {
-    await page.typeIntoElement(this.cphNumberInput, text)
+    await page.typeIntoElement(await this.cphNumberInput(), text)
```
Comment (hughfdjackson) [inline test/page-objects/parishHoldingNumberPage.js:36]: No need for extra await here

File: test/page-objects/parishHoldingNumberPage.js [hunk 7]
```diff
@@ -12,37 +12,40 @@ class ParishHoldingNumberPage extends Page {
...
     await super.selectContinue()
   }
 
   async parishHoldingErrorTest(textInput, errorMessage) {
     await this.inputParishHoldingNumberAndContinue(textInput)
-    await super.verifyErrorsOnPage(this.cphInputFieldError, errorMessage)
+    await super.verifyErrorsOnPage(
+      await this.cphInputFieldError(),
```
Comment (hughfdjackson) [inline test/page-objects/parishHoldingNumberPage.js:43]: No need for extra await here either :D

File: test/page-objects/parishHoldingNumberPage.js (hunk 5 above)
Comment (hughfdjackson) [reply test/page-objects/parishHoldingNumberPage.js:15]: Same for all other methods that are calling `super.getInputField(....)` and returning its result only

**Review by hughfdjackson — CHANGES_REQUESTED** [review]

**Review by hughfdjackson — APPROVED** [review]

**Pull request merged**